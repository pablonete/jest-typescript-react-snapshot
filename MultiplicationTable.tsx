import * as React from 'react';

export interface Props {
  factor: number;
}

export const MultiplicationTable = (props: Props): JSX.Element =>
  <ul>
    { getRange(1, 10).map(i => 
      <li key={i}>{props.factor} x {i} = {props.factor*i}</li>
    )}
  </ul>;

function getRange(from: number, to: number): number[] {
  return Array.apply(null, Array(to - from + 1)).map((_, i) => i + from);
}