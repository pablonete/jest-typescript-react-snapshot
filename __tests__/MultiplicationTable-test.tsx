import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MultiplicationTable } from "../MultiplicationTable";

it('MultiplicationTable for number 5', () => {
  const checkbox = renderer.create(
    <MultiplicationTable factor={5} />
  );

  expect(checkbox.toJSON()).toMatchSnapshot();
});
