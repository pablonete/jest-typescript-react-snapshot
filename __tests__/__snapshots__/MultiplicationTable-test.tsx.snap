exports[`test MultiplicationTable for number 5 1`] = `
<ul>
  <li>
    5
     x 
    1
     = 
    5
  </li>
  <li>
    5
     x 
    2
     = 
    10
  </li>
  <li>
    5
     x 
    3
     = 
    15
  </li>
  <li>
    5
     x 
    4
     = 
    20
  </li>
  <li>
    5
     x 
    5
     = 
    25
  </li>
  <li>
    5
     x 
    6
     = 
    30
  </li>
  <li>
    5
     x 
    7
     = 
    35
  </li>
  <li>
    5
     x 
    8
     = 
    40
  </li>
  <li>
    5
     x 
    9
     = 
    45
  </li>
  <li>
    5
     x 
    10
     = 
    50
  </li>
</ul>
`;
